src/Generator/Generators/CLI/CLIHeadersTemplate.cs

```diff
@@ -137,6 +137,9 @@ public void GenerateForwardRefs()
             var typeReferences = typeReferenceCollector.TypeReferences;
             var @namespace = ConvertForwardReferencesToNamespaces(typeReferences);
 
+            foreach (var cls in @namespace.Classes)
+                cls.IsStruct = ExtensionMethods.CheckClassIsStructible(cls, Driver);
```

Comment:
I think you'd best move the whole logic about checking whether a type can be represented as a struct, to a new pass.